platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy

```diff
@@ -276,7 +277,13 @@ class ToolingApi implements TestRule {
 
     private createConnector() {
         if (isolatedToolingClient != null) {
-            return isolatedToolingClient.getFactory(DefaultGradleConnector).create()
+            // This fixture can be loaded with a classloader of TAPI jar from previous Gradle releases
+            def currentVersion = GradleVersion.current().baseVersion
+            if (currentVersion <= GradleVersion.version("8.13")) {
```

Comment:
🤔  Instead of introducing yet another variation over versions we should probably introduce an indirection to make it more future-proof.

It probably then makes more sense to do this in 9.0, because it will require a sort of internal API that we want to keep cross-version stable.